src/runtime/hexagon_remote/halide_host.cpp

```diff
@@ -116,19 +117,18 @@ __attribute__((weak)) void remote_register_buf(void* buf, int size, int fd);
 
 void halide_hexagon_host_malloc_init() {
     pthread_mutex_init(&allocations_mutex, NULL);
+    ion_fd = open("/dev/ion", O_RDONLY, 0);
```

Comment:
Oh, I see what you mean. When I was looking at this before, open(/dev/ion) always seemed to return the same FD (as long as the ref count didn't go to zero). But, that seems like a bad thing to rely on, I'll rework this code.